### PR TITLE
move api/errors to apimachinery

### PR DESCRIPTION
**Pull request by deads2k**
`pkg/api/errors` is a set of helpers around `meta/v1.Status` that help to create and interpret various apiserver errors.  Things like `.NewNotFound` and `IsNotFound` pairings.  This pull moves it into apimachinery for use by the clients and servers.

@smarterclayton @lavalamp First commit is the move plus minor fitting.  Second commit is straight replace and generation.

### CODE REVIEW

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ import (
 	"sync"
 	"time"
 
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
```
Comment (sttts) [inline test/e2e/framework/framework.go]: apierrors

File: pkg/api/errors/doc.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2014 The Kubernetes Authors.
+Copyright 2016 The Kubernetes Authors.
```
Comment (sttts) [inline pkg/api/errors/doc.go]: 2017

**Comment by sttts** [comment]
Some nits.

Otherwise, I am fine with the move. Certainly, k8s.io/apimachinery is not strictly pure apimachinery anymore, but contains some other code which is somehow near the apis _and_ shared between server and client. IMO that's ok as it is pragmatic for now. If k8s.io/apimachinery grow too much, we can still think about splitting it up further.

**Comment by deads2k** [comment]
@k8s-bot kubemark e2e test this

**Pull request merged**